spec/monthly_inventory_test.rb

```diff
@@ -0,0 +1,138 @@
+# frozen_string_literal: true
+
+require "date"
+require "fileutils"
+require_relative "../lib/monthly_inventory"
+
+HOME = File.expand_path(File.join(__dir__, "..")).freeze
+TEST_TMP = File.join(HOME, "monthly_inventory_test_temp").freeze
+
+CATALOG_PREP = File.join(TEST_TMP, "catalog_prep").freeze
+RIGHTS_DIR = File.join(TEST_TMP, "rights").freeze
+RIGHTS_DIR_ARCHIVE = File.join(TEST_TMP, "rights", "archive").freeze
+INGEST_BIBRECORDS = File.join(TEST_TMP, "ingest_bibrecords").freeze
+INGEST_BIBRECORDS_ARCHIVE = File.join(TEST_TMP, "ingest_bibrecords", "archive").freeze
+
+LOCATIONS = [
+  CATALOG_PREP,
+  RIGHTS_DIR,
+  RIGHTS_DIR_ARCHIVE,
+  INGEST_BIBRECORDS,
+  INGEST_BIBRECORDS_ARCHIVE
+].freeze
+
+ENV["CATALOG_PREP"] = CATALOG_PREP
+ENV["RIGHTS_DIR"] = RIGHTS_DIR
+ENV["INGEST_BIBRECORDS"] = INGEST_BIBRECORDS
+puts "ENV[\"CATALOG_PREP\"] = \"#{CATALOG_PREP}\""
+puts "ENV[\"RIGHTS_DIR\"] = \"#{RIGHTS_DIR}\""
+puts "ENV[\"INGEST_BIBRECORDS\"] = \"#{INGEST_BIBRECORDS}\""
+# Create monthly_inventory_test_temp directory
+# Create temp CATALOG_PREP, RIGHTS_DIR, INGEST_BIBRECORDS if necessary
+# Doing these tests in a separate directory that can be thrown away to
+# avoid stepping on the data/ directory before we can get this new Ruby
+# code better integrated.
+def setup_test_tmp
+  Dir.mkdir(TEST_TMP) unless File.directory? TEST_TMP
+  LOCATIONS.each do |loc|
+    Dir.mkdir(loc) unless File.directory?(loc)
+  end
+end
+
+def update_file_for_date(date:)
+  File.join(
+    CATALOG_PREP,
+    "zephir_upd_YYYYMMDD.json.gz".sub("YYYYMMDD", date.strftime("%Y%m%d"))
```

Comment:
I like having that very readable date template -- since there are unhyphenated HT files and hyphenated Zephir files I've tried to highlight difference -- but with the addition of a warning comment I have adopted this succinct variant.